features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/prerequisite/SessionReplayRequirement.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.prerequisite
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.utils.submitSafe
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+internal class SessionReplayRequirement private constructor(
+    private val checkers: List<SystemRequirementChecker>,
+    private val systemRequirementConfiguration: SystemRequirementConfiguration
+) {
+    private var executorService: ExecutorService = Executors.newSingleThreadExecutor()
```

Comment:
We shouldn't spin new thread if it is possible to re-use existing ones. Here we create a thread for a single operation and then it is not needed. It is possible to use a thread from some existing pool?

Reply:
I don't see any shared thread pool across the modules in our sdk, do you have any suggestion?